common/protocol/http/sniff.go

```diff
@@ -65,6 +65,10 @@ func SniffHTTP(b []byte) (*SniffHeader, error) {
 		version: HTTP1,
 	}
 
+	if !shouldSniffDomain {
```

Comment:
I understand the intention here. However, at this point, the request content is already read in memory. It doesn't seem very expensive to, parse a http header, for example.

Reply:
this do skip the process of parsing host.

while in current process, domain sniffing of FakeDNS makes nothing, it's simply duplicated.